src/interpreter/map-validator.ts

```diff
@@ -375,8 +380,10 @@ export class MapValidator implements MapAstVisitor {
       // Init new variables if object used in dot notation wasn't defined before
       if (assignment.key.length > 1) {
         const keys = [];
-        for (const key of assignment.key) {
-          keys.push(key);
+        for (let i = 0; i < assignment.key.length - 1; i++) {
```

Comment:
Why did you change this? I

Reply:
In old approach, it initializes even whole variable, which is then overwritten because it's assignment anyway.

For example for `some.variable.key = "string"` I wanted to initialize only `some`, `some.variable` and let assignment node handle storing `some.variable.key`.